File: csharp/Structural/09-Decorator/Decorator/WarrantiedDecorator.cs
using Decorator.Component;

namespace Decorator.Decorator;

public class WarrantiedDecorator : JobCardDecorator
{
    private const int WarrantyDays = 90;

    public WarrantiedDecorator(IJobCard jobCard) : base(jobCard) { }

    public override string Description =>
        $"🔒 WARRANTIED ({WarrantyDays}d) | {_wrapped.Description}";

    public override void Execute()
    {
        base.Execute();
        Console.WriteLine($"  📆 90-day callback warranty registered for Job #{JobId}.");
    }
}

File: typescript/Structural/09-Decorator/Decorator/WarrantiedDecorator.ts
import { IJobCard } from "../Component/IJobCard";
import { JobCardDecorator } from "./JobCardDecorator";

export class WarrantiedDecorator extends JobCardDecorator {
  private static readonly WARRANTY_DAYS = 90;

  constructor(jobCard: IJobCard) {
    super(jobCard);
  }

  get description(): string {
    return `🔒 WARRANTIED (${WarrantiedDecorator.WARRANTY_DAYS}d) | ${this.wrapped.description}`;
  }

  execute(): void {
    super.execute();
    console.log(
      `  📆 90-day callback warranty registered for Job #${this.jobId}.`,
    );
  }
}
